app/controllers/concerns/sign_in_duration_concern.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+module SignInDurationConcern
+  extend ActiveSupport::Concern
+
+  def sign_in_duration_seconds
+    return unless session[:sign_in_page_visited_at]
+    (Time.zone.now - Time.zone.parse(session[:sign_in_page_visited_at])).seconds.to_i
```

Comment:
I guess I'm trying to figure out how to do that math without flipping around between a TimeWithZone and a string.

If I don't  change to a string I get this error
ArgumentError:
    invalid value for convert():
    
Possibly the time difference could be ascertained in a smoother way...

Reply:
Ohhh, I see now, we assign `Time.zone.now` (a date), but it gets serialized into the session as a string. In that case, I think it makes sense that we stub the test code with a string as well.

We could also make this a bit more explicit by casting `Time.zone.now` to a string (`Time.zone.now.to_s`) at the time it's assigned into the session, to avoid any confusion about what type it is.